metrics/transport.go

```diff
@@ -88,7 +89,12 @@ func PushMetrics(ctx context.Context, metricsPort int, metricsAddress, job strin
 		return err
 	}
 
-	var addr = flag.String("push-address", fmt.Sprintf("%s:%d", metricsAddress, metricsPort), "The address to push prometheus metrics")
+	u, err := url.Parse(metricsAddress)
+	if err != nil {
+		panic(err)
```

Comment:
Is this the behaviour we want? If we can't parse a push gateway (and thus send metrics) don't execute the test at all?

Reply:
Altered the code to print and error (stderr) and ignore trying to push metrics